Sources/PublicKey.swift

```diff
@@ -38,7 +38,7 @@ public struct PublicKey: Hashable, CustomStringConvertible {
         if isCompressed {
             return self
         }
-        let prefix: UInt8 = 0x02 | (data[1] & 0x01)
+        let prefix: UInt8 = 0x02 | (data[64] & 0x01)
```

Comment:
from:
https://bitcoin.stackexchange.com/questions/76181/bitcoin-public-keys-from-uncompressed-to-compressed-version-with-code-sample/76183#76183
`Compression is very simply, if the Y value of the pubkey is odd, place an 0x03 byte. If it is even, place a 0x02 byte`

And we been using  0x02 and 0x01, can you explain this part?

Reply:
0.  `&` `|` are bitwise operators, here is for fast computing
1. `data[64] & 0x01` test if y is even or odd (you convert it to binary and do bitwise AND)
2. `0x02 |` does bitwise OR 0 / 1 (y) and results 0x02 | 0x03.